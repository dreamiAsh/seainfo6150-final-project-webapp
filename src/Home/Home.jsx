import React, {useEffect, useState} from "react";
import treefrog from "../images/treefrog.jpg";
import styles from "./Home.module.css";
import NewsList from "../Components/NewsList/NewsList"
import RecipeItem from "../Components/RecipeItem/RecipeItem"
import RecipeList from "../Components/RecipeList/RecipeList"

const Home = () => {

  // const [fetchedData, setFetchedData] = useState({});
  // useEffect(() => {
  //   const fetchData = async () => {
  //     // put data fetching code here!
  //     // performs a GET request
  //     const response = await fetch(
  //       "http://demo9533681.mockable.io/recipeList"
  //     );
  //     const responseJson = await response.json();
  //     setFetchedData(responseJson);
  //   };
  //   fetchData();
  // }, [fetchedData]);

  //   const myList = Object.values(fetchedData)
  //   const mappedList = myList.map((listItem) => (
  //     <RecipeItem recipe={listItem}/>
  //   ))

    return (
        <div>
            <div className={styles.imgBox}>
                <img className={styles.img} src="https://images.squarespace-cdn.com/content/v1/5dedf21b7c131a3015fd5ad5/1575948055123-VLZ49JEXIYF8QMAABP6F/ke17ZwdGBToddI8pDm48kGNd1IaaP_-ea7nYNEFHDsp7gQa3H78H3Y0txjaiv_0fDoOvxcdMmMKkDsyUqMSsMWxHk725yiiHCCLfrh8O1z5QPOohDIaIeljMHgDF5CVlOqpeNLcJ80NK65_fV7S1UUIsyChEB7CNmEcY1NH_HD7WSuQ7QDejvk14LXDmYznTAN43Xjxxvi5UJclDbUAqLw/DSC04905-43_web.jpg?format=2500w" alt="food poster"/>
            </div>
            <NewsList/>
            <RecipeList/>
        </div>
    )
}

export default Home
