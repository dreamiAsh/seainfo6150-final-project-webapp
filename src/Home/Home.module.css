*{margin: 0;padding: 0;list-style: none;}
.h1{
    color: yellow;
}
.imgBox{
    width: 100%;
    height: 350px;
    overflow: hidden;}
.img{
    /* /* width: 100%; */
    width: 100%;
}
.recipeList{
    width: 85%;
    height: 800px;
    text-align: center;
    padding: 0 7%;
}