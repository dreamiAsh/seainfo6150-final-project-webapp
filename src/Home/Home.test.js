import React from "react";
import { render, fireEvent } from "@testing-library/react";
import Home from "./Home.jsx";

it("renders the home component correctly", () => {
  const { container } = render(<Home />);
  expect(container).toMatchSnapshot();
});
