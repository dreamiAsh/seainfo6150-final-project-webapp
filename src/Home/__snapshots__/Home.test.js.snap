// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the home component correctly 1`] = `
<div>
  <div>
    <div
      class="imgBox"
    >
      <img
        alt="food poster"
        class="img"
        src="https://images.squarespace-cdn.com/content/v1/5dedf21b7c131a3015fd5ad5/1575948055123-VLZ49JEXIYF8QMAABP6F/ke17ZwdGBToddI8pDm48kGNd1IaaP_-ea7nYNEFHDsp7gQa3H78H3Y0txjaiv_0fDoOvxcdMmMKkDsyUqMSsMWxHk725yiiHCCLfrh8O1z5QPOohDIaIeljMHgDF5CVlOqpeNLcJ80NK65_fV7S1UUIsyChEB7CNmEcY1NH_HD7WSuQ7QDejvk14LXDmYznTAN43Xjxxvi5UJclDbUAqLw/DSC04905-43_web.jpg?format=2500w"
      />
    </div>
    <div
      class="box"
    >
      <div
        class="leftBox"
      >
        <h4
          class="itemTitle"
        >
          What's new
        </h4>
        <ul />
      </div>
      <div
        class="rightBox"
      >
        <h4
          class="itemTitle"
        >
          Most popular
        </h4>
        <ul />
      </div>
    </div>
    <ul
      class="recipeList"
    />
  </div>
</div>
`;
